.github/workflows/publish_pytorch_dev_docker.yml

```
@@ -72,3 +76,69 @@ jobs:
           push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+      - name: Print image digest
+        run: |
+            echo "Digest: ${{ steps.build_push.outputs.digest }}"
+  verify-image:
+    needs: build-and-push-image
+    strategy:
+      matrix:
+        targets:
+          - amdgpu_target: "gfx942"
+    uses: ./.github/workflows/verify_docker_image.yml
+    with:
+      amdgpu_target: ${{  matrix.targets.amdgpu_target }}
+      image_digest: ${{ needs.build-and-push-image.outputs.image_digest }}
+
+  on-success:
+    needs: verify-image
+    if: ${{ needs.verify-image.result == 'success' }}
+    strategy:
+      matrix:
+        targets:
+          - amdgpu_target: "gfx942"
+    runs-on: azure-linux-scale-rocm
+    steps:
+      - name: Log in to the Container registry
+        uses: docker/login-action@9780b0c442fbb1117ed29e0efdff1e18412f7567 # v3.3.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Write pinned version to file
+        run: |
+          echo "${{ env.VERSION }}" > latest_verified_image_version_pinned.txt
+      - name: Upload pinned version file as an artifact
+        uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # v4.6.1
+        with:
+          name: latest_verified_image_version_pinned-${{ matrix.targets.amdgpu_target }}
+          path: latest_verified_image_version_pinned.txt
+      - name: Tag and Push 'verified'
+        run: |
+          IMAGE=ghcr.io/rocm/therock_pytorch_dev_ubuntu_24_04_${{ matrix.targets.amdgpu_target }}
+          DIGEST=${{ needs.verify-image.outputs.image_digest }}
```

```suggestion
          DIGEST=${{ needs.build-and-push-image.outputs.image_digest }}
```

Shouldn't it be the digest created in the build and push step?